devdoc/waiter_queue_requirements.md

```diff
@@ -0,0 +1,103 @@
+# `waiter_queue` requirements
+
+## Overview
+
+`waiter_queue` is a module that allows consumers to park calls and for producers to signal that the parked calls be executed. A "parked call" consists of a callback function and a context. The consumer provides the callback function and context when it parks a call. The consumer expects the parked call's callback function to be executed with the provided context when the producer signals.
+
+`waiter_queue` has two parts: `waiter_queue_ll` that contains the logic for parking and unblocking calls and `waiter_queue_hl` that contains logic for making `waiter_queue` thread-safe.
+
+Consumers call `waiter_queue_hl_add_waiter` with a `UNBLOCK_CALLBACK` callback function and context to park a call. The parked call is added to a queue and will be executed when the producers signals by calling `waiter_queue_ll_unblock_waiters`.
```

Comment:
>waiter_queue_ll_unblock_waiters [](http://example.com/codeflow?start=212&length=31)

This sounds a bit weird as the consumers exists in the HL layer (waiter_queue_*hl*_add_waiter) and the producer are in LL layer (waiter_queue_*ll*_unblock_waiters)

Reply:
Typo. That should be `waiter_queue_hl_unblock_waiters`